packages/aws/src/RPA/Cloud/AWS/__init__.py

```diff
@@ -990,7 +1024,672 @@ def delete_queue(self, queue_name: str = None):
+            or more characters and "_" means match any one character.
+            Only schema name entries matching the search pattern are returned.
+            If ``schema_pattern`` is not specified, then all schemas are returned.
+        """
+        client = self._get_client_for_service("redshift-data")
+        additional_params = self._create_auth_params(database)
+        if schema_pattern:
+            additional_params["SchemaPattern"] = schema_pattern
+        paginator = client.get_paginator("list_schemas")
+        return paginator.paginate(**additional_params).build_full_result()["Schemas"]
+
+    def _create_auth_params(self, alternate_database: Optional[str] = None) -> Dict:
+        """Generates a dictionary of authentication params depending
+        on which method was defined when initializing this class. It should
+        be called before adding call-specific parameters. If an alternate
+        database name is provided, this function checks if it matches the
+        configured database name, and if it does not, adds the params
+        `ConnectedDatabase` and `Database`, otherwise it only adds `Database`.
+        """
+        auth_params = {}
+        if self.cluster_identifier:
+            auth_params["ClusterIdentifier"] = self.cluster_identifier
+        if self.secret_arn:
+            auth_params["SecretArn"] = self.secret_arn
+        elif self.database_user:
+            auth_params["DbUser"] = self.database_user
+        if alternate_database == self.database or not alternate_database:
+            auth_params["Database"] = self.database
+        else:
+            auth_params["ConnectedDatabase"] = self.database
+            auth_params["Database"] = alternate_database
+        return auth_params
+
+
+class ServiceSTS(AWSBase):
+    """Class for AWS STS Service."""
+
+    def __init__(self) -> None:
+        self.services.append("sts")
+        self.logger.debug("ServiceSts init")
+
+    def init_sts_client(
+        self,
+        aws_key_id: str = None,
+        aws_key: str = None,
+        region: str = None,
+        use_robocloud_vault: bool = False,
+        session_token: str = None,
+    ) -> None:
+        """Initialize AWS STS client.
+
+        :param aws_key_id: access key ID
+        :param aws_key: secret access key
+        :param region: AWS region
+        :param use_robocloud_vault: use secret stored into `Robocloud Vault`
+        :param session_token: a session token associated with temporary
+            credentials, such as from ``Assume Role``.
+        """
+        self._init_client(
+            "sts", aws_key_id, aws_key, region, use_robocloud_vault, session_token
+        )
+
+    @aws_dependency_required
+    def assume_role(
+        self,
+        role_arn: str,
+        role_session_name: str,
+        policy_arns: Optional[List[Dict]] = None,
+        policy: Optional[str] = None,
+        duration: Optional[int] = 900,
```

Comment:
```suggestion
        duration: int = 900,
```
**if** shouldn't be ever set to `None` (null in the payload). Which should be fine since you're not providing the keyword in the params if the value doesn't evaluate to `True`, but what if we set a duration of `0` (which evaluates as `False` at line 1678)? -- which in this case isn't added and should be the right thing to do

Reply:
Good point, needed to define a minimum anyway.